sphinx_gallery/scrapers.py

```diff
@@ -128,44 +131,24 @@ def matplotlib_scraper(block, block_vars, gallery_conf, **kwargs):
 
     image_path_iterator = block_vars["image_path_iterator"]
     image_rsts = []
-    # Check for srcset hidpi images
     srcset = gallery_conf["image_srcset"]
-    srcset_mult_facs = [1]  # one is always supplied...
-    for st in srcset:
-        if (len(st) > 0) and (st[-1] == "x"):
-            # "2x" = "2.0"
-            srcset_mult_facs += [float(st[:-1])]
-        elif st == "":
-            pass
-        else:
-            raise ExtensionError(
-                f'Invalid value for image_srcset parameter: "{st}". '
-                "Must be a list of strings with the multiplicative "
-                'factor followed by an "x".  e.g. ["2.0x", "1.5x"]'
-            )
 
     # Check for animations
-    anims = list()
+    anims = {}
     if gallery_conf["matplotlib_animations"]:
         for ani in block_vars["example_globals"].values():
             if isinstance(ani, Animation):
-                anims.append(ani)
+                anims[ani._fig] = ani
```

Comment:
I worry about using the `._fig` private attribute here

Reply:
That's what the original code used; it's just been moved.